cocotb/scheduler.py

```diff
@@ -453,24 +451,13 @@ def _event_loop(self, trigger):
                     if _debug:
                         self.log.debug("Scheduled coroutine %s" % (coro._coro.__qualname__))
 
-                    # remove our reference to the objects at the end of each loop,
-                    # to try and avoid them being destroyed at a weird time (as
-                    # happened in gh-957)
-                    del coro
-
```

Comment:
This `del`, and many of the others, is important.

Reply:
@eric-wieser reverted the changes.